src/main/scala/scorex/core/api/http/PeersApiRoute.scala

```diff
@@ -36,14 +37,19 @@ case class PeersApiRoute(peerManager: ActorRef,
   }
 
   def connectedPeers: Route = (path("connected") & get) {
-    val result = askActor[Seq[PeerInfo]](networkController, GetConnectedPeers).map {
-      _.map { peerInfo =>
-        PeerInfoResponse(
-          address = peerInfo.peerSpec.declaredAddress.map(_.toString).getOrElse(""),
-          lastSeen = peerInfo.lastSeen,
-          name = peerInfo.peerSpec.nodeName,
-          connectionType = peerInfo.connectionType.map(_.toString)
-        )
+    val result = askActor[Seq[ConnectedPeer]](networkController, GetConnectedPeers).map {
+      _.map { con =>
+        con.peerInfo match {
+          case Some(peerInfo) =>
+            PeerInfoResponse(
+              address = peerInfo.peerSpec.declaredAddress.map(_.toString).getOrElse(""),
+              lastMessage = con.lastMessage,
+              lastHandshake = peerInfo.lastHandshake,
+              name = peerInfo.peerSpec.nodeName,
+              connectionType = peerInfo.connectionType.map(_.toString)
+            )
+          case None => throw new Exception("Peer is not connected")
```

Comment:
We shouldn't get here, but throwing exceptions in API is not good, log.error is enough (thus connected peers will be returned)

Reply:
Sorry, I do not know how to replace throw with just logging (because otherwise  compiler reports error).
Actually I need to call "get", because Option is this case is guaranteed to be non empty. But using Option.get is now prohibited. May be you can suggest some other better to handle this situation in Scala (transform components of a sequence, ignoring elements which Option fields are empty).

Reply:
ah I see, replaced .map with flatmap, and not checking errors at all (such checks should be done in other places)